app/assets/javascripts/image-upload-markdown.js

```diff
@@ -0,0 +1,66 @@
+// This is a modified version of https://github.com/stuartnelson3/blog/blob/master/public/js/script.js
+
+$(document).on('drop', '.js-markdown-image-upload', function(e) {
+  e.preventDefault();
+  var $textarea = $(this);
+
+  var dt = e.originalEvent.dataTransfer;
+
+  addUploadPlaceholders($textarea, dt);
+  for (var i = 0; i < dt.files.length; i++) {
+    var file = dt.files[i];
+    handleFileUpload(file);
+  }
+
+  function addUploadPlaceholders($textarea, dataTransfer) {
+    var placeholders = [];
+    for (var i = 0; i < dataTransfer.files.length; i++) {
+      var filename = dataTransfer.files[i].name;
+      placeholders.push(placeholderValue(filename));
+    }
+
+    var selectionStart = $textarea[0].selectionStart;
+    var uploadsPlaceholder = placeholders.join("\r\n");
+    $textarea.val(
+      [$textarea.val().slice(0, selectionStart), uploadsPlaceholder, $textarea.val().slice(selectionStart)].join('')
+    );
+  }
+
+  function replaceImageMarkdownPlaceholder($textarea, newValue, filename){
+    var oldPlaceholder = placeholderValue(filename);
+    $textarea.val(
+      $textarea.val().replace(oldPlaceholder, newValue)
+    );
+  }
+
+  function placeholderValue(filename) {
```

Comment:
There are other places you could use this function, if you changed the signature to `(filename, filepath)` ;)